lib/client/rpc/proposals.ts

```diff
@@ -242,7 +251,184 @@ export class FutarchyRPCProposalsClient implements FutarchyProposalsClient {
+    tx.recentBlockhash = blockhash;
+    tx.feePayer = this.rpcProvider.wallet.publicKey; // Assuming the provider has a wallet
+
+    try {
+
+      const simulationResult = this.rpcProvider.connection.simulateTransaction(tx)
+      console.log(simulationResult)
+    } catch (e) {
+      console.error("Simulation error:", e)
+    }
+    // const wireTransaction = tx.
+  }
+
+  async getUserTokenBalance(tokenMintAddress: PublicKey) {
+    // Get the associated token account for the user's wallet and the token mint
+    const userTokenAccountAddress = await getAssociatedTokenAddress(
+      new PublicKey(tokenMintAddress),
+      this.rpcProvider.publicKey
+    );
+
+    // Fetch the account info
+    const accountInfo = await getAccount(this.rpcProvider.connection, userTokenAccountAddress);
+
+    // The amount is a BigInt, convert it to a string if necessary
+    return accountInfo.amount.toString();
+  }
+
+  async mintIx(amount: number,
+    vaultAccountAddress: PublicKey,
+    vaultAccount: VaultAccountWithProtocol) {
+    const ixs = [
+      createAssociatedTokenAccountIdempotentInstruction(
+        this.rpcProvider.publicKey,
+        getAssociatedTokenAddressSync(
+          vaultAccount.conditionalOnFinalizeTokenMint,
+          this.rpcProvider.publicKey,
+          true
+        ),
+        this.rpcProvider.publicKey,
+        vaultAccount.conditionalOnFinalizeTokenMint
+      ),
+      createAssociatedTokenAccountIdempotentInstruction(
+        this.rpcProvider.publicKey,
+        getAssociatedTokenAddressSync(
+          vaultAccount.conditionalOnRevertTokenMint,
+          this.rpcProvider.publicKey,
+          true
+        ),
+        this.rpcProvider.publicKey,
+        vaultAccount.conditionalOnRevertTokenMint
+      ),
+      await vaultAccount.protocol.vault.methods
+        .mintConditionalTokens(amount)
+        .accounts({
+          vault: vaultAccountAddress,
+          userConditionalOnFinalizeTokenAccount: getAssociatedTokenAddressSync(
+            vaultAccount.conditionalOnFinalizeTokenMint,
+            this.rpcProvider.publicKey,
+            true
+          ),
+          userConditionalOnRevertTokenAccount: getAssociatedTokenAddressSync(
+            vaultAccount.conditionalOnRevertTokenMint,
+            this.rpcProvider.publicKey,
+            true
+          ),
+          userUnderlyingTokenAccount: getAssociatedTokenAddressSync(
+            vaultAccount.underlyingTokenMint,
+            this.rpcProvider.publicKey,
+            true
+          ),
+          vaultUnderlyingTokenAccount: vaultAccount.underlyingTokenAccount,
+          conditionalOnFinalizeTokenMint:
+            vaultAccount.conditionalOnFinalizeTokenMint,
+          conditionalOnRevertTokenMint:
+            vaultAccount.conditionalOnRevertTokenMint,
+        })
+        .instruction(),
+    ];
+    return ixs
+  }
+
+  async createProposal(daoAggregate: DaoAggregate, user: PublicKey) {
```

Comment:
I don't think we'll be able to just pass in the DaoAggregate because we need to know which autocrat to actually create the proposal on, and the DaoAggregate spans across multiple autocrats.

Reply:
we check under it if the daoaggregate contains a dao that has the proper version, if not we don't go through

Reply:
Might be great to have a version selector but as v0.2 is mostly to be sure we can do it more than to use it, I didn't do the 100% graceful one